src/Identity/Core/src/IdentityBuilderExtensions.cs

```diff
@@ -41,6 +41,7 @@ public static IdentityBuilder AddDefaultTokenProviders(this IdentityBuilder buil
     private static void AddSignInManagerDeps(this IdentityBuilder builder)
     {
         builder.Services.AddHttpContextAccessor();
+        builder.Services.AddScoped<IPasskeyRequestContextProvider, HttpPasskeyRequestContextProvider>();
         builder.Services.AddScoped(typeof(ISecurityStampValidator), typeof(SecurityStampValidator<>).MakeGenericType(builder.UserType));
         builder.Services.AddScoped(typeof(ITwoFactorSecurityStampValidator), typeof(TwoFactorSecurityStampValidator<>).MakeGenericType(builder.UserType));
```

Comment:
This service allows passkey handlers to infer the server's origin from the ongoing request. We can't register this by default in `Microsoft.Extensions.Identity.Core` because that package can't assume an `HttpContext` will be available. Putting the service registration here means that it will get registered as long as the app is already calling `AddSignInManager()`.

However, we might be able to eliminate the need for this abstraction completely by adjusting the layering and moving most of the passkey logic to `Microsoft.AspNetCore.Identity`. The `M.E.Identity.Core` package is mostly web-agnostic, but WebAuthn is very much not, so it might make sense to move things around.

Reply:
Done.